### PR TITLE
remove oc cluster up dependency on oc binary

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
this is un-verifiable until https://github.com/openshift/origin/issues/15381 is resolved.

**Comment by bparees** [comment]
@csrwng ptal

**Comment by csrwng** [comment]
> this is un-verifiable until #15381 is resolved.

can't it be verified with the released version of the images?

File: pkg/bootstrap/docker/openshift/helper.go [hunk 1]
```diff
@@ -777,32 +779,34 @@ func (h *Helper) updateConfig(configDir string, opt *StartOptions) error {
...
+			SignerCertOptions: signerCertOptions,
+			ClientDir:         configDir,
+			ExpireDays:        crypto.DefaultCertificateLifetimeInDays,
+			User:              "aggregator-front-proxy",
+			Output:            cmdOutput,
+		}
+		createClientOptions.CreateClientFolder()
 		if err != nil {
-			return errors.NewError(fmt.Sprintf("failed generating client certificate, command output: %s\nerror: %v", out, err))
+			return errors.NewError("cannot create client config").WithCause(err)
```
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/helper.go]: Add the command output as detail to the error:
```
errors.NewError("cannot create client config").WithCause(err).WithDetail(cmdOutput)
```

File: pkg/bootstrap/docker/openshift/helper.go [hunk 2]
```diff
@@ -777,32 +779,34 @@ func (h *Helper) updateConfig(configDir string, opt *StartOptions) error {
...
-			"adm",
-			"create-api-client-config",
-			"--certificate-authority", cacertPath,
-			"--signer-cert", cacertPath,
-			"--signer-key", cakeyPath,
-			"--signer-serial", caserialPath,
-			"--user", "aggregator-front-proxy",
-			"--client-dir", configDir,
-		).CombinedOutput()
+			return errors.NewError("cannot create signer cert").WithCause(err)
```
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/helper.go]: Same as below, add the output to the error details

**Comment by bparees** [comment]
> can't it be verified with the released version of the images?

technically perhaps, but i'd feel better seeing it work end to end w/ the latest origin images.

File: pkg/bootstrap/docker/openshift/helper.go [hunk 3]
```diff
@@ -777,32 +779,34 @@ func (h *Helper) updateConfig(configDir string, opt *StartOptions) error {
...
+		signerCertOptions.SerialFile = caserialPath
+
+		createClientOptions := &admin.CreateClientOptions{
+			SignerCertOptions: signerCertOptions,
+			ClientDir:         configDir,
+			ExpireDays:        crypto.DefaultCertificateLifetimeInDays,
+			User:              "aggregator-front-proxy",
+			Output:            cmdOutput,
+		}
+		createClientOptions.CreateClientFolder()
```
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/helper.go]: looks like you missed assigning the err variable

File: pkg/bootstrap/docker/openshift/helper.go (hunk 2 above)
Comment (bparees) [inline pkg/bootstrap/docker/openshift/helper.go]: "do as i say, not as i did", eh? https://github.com/openshift/origin/blob/master/pkg/bootstrap/docker/openshift/admin.go#L136-L157

Comment (csrwng) [inline pkg/bootstrap/docker/openshift/helper.go]: It's called thought evolution :)

File: pkg/bootstrap/docker/openshift/helper.go (hunk 3 above)
Comment (bparees) [inline pkg/bootstrap/docker/openshift/helper.go]: doh.

**Comment by bparees** [comment]
@csrwng updated.  does seem to work with 3.6.0

**Comment by bparees** [comment]
@smarterclayton @stevekuznetsov so at the time the bot posted this "failure" message (midnight) the test job in question was still running....

**Comment by bparees** [comment]
I guess it's in tear-down mode.

**Comment by LalatenduMohanty** [comment]
@bparees @csrwng Will this PR be part of 3.6 release ? Preferably 3.6 RC1.

**Comment by stevekuznetsov** [comment]
@bparees yeah we publish failure result as the first post-run step, whereas the rest of cleanup happens after

**Comment by stevekuznetsov** [comment]
Also technically the Jenkins job is in the failed state as soon as any build steps fail, and cannot leave the failed state as a result of post-build execution. With Rosie we always updated GitHub right before the job finished, but that's not the case anymore.

**Comment by bparees** [comment]
@LalatenduMohanty was not planning on it, but yes I can backport it for you.

**Comment by bparees** [comment]
backport is here: https://github.com/openshift/origin/pull/15471

**Comment by LalatenduMohanty** [comment]
@bparees Awesome thanks.

**Pull request merged**